r/vignettes/install.Rmd

```diff
@@ -379,6 +330,13 @@ By default, these are all unset. All boolean variables are case-insensitive.
   The directory will be created if it does not exist.
 * `CMAKE`: When building the C++ library from source, you can specify a
   `/path/to/cmake` to use a different version than whatever is found on the `$PATH`
+* `ARROW_S3`: If set to `true` S3 support will be built (as long as the 
```

Comment:
Let's make this a separate bullet list since these are features to be enabled/disabled in the cmake. Note also what the defaults are. Also might be good to enumerate the compression libraries (could group them all under `ARROW_WITH_*` or something).

Reply:
I also moved them up to the top of this section, I think these are actually slightly more interesting / important than the other group